connectors/http/http-base/src/main/java/io/camunda/connector/http/base/client/apache/ProxyHandler.java

```diff
@@ -0,0 +1,121 @@
+import org.apache.hc.client5.http.impl.auth.BasicCredentialsProvider;
+import org.apache.hc.client5.http.impl.routing.DefaultProxyRoutePlanner;
+import org.apache.hc.client5.http.impl.routing.SystemDefaultRoutePlanner;
+import org.apache.hc.client5.http.routing.HttpRoutePlanner;
+import org.apache.hc.core5.http.HttpHost;
+
+public class ProxyHandler {
+
+  record CustomProxy(String host, int port, String username, String password) {}
+
+  public static CredentialsProvider getCredentialsProvider(String uncheckedProtocol) {
+    BasicCredentialsProvider provider = new BasicCredentialsProvider();
+    String protocol = fallbackOnHttpForInvalidProtocol(uncheckedProtocol);
+    try {
+      if (isValidProxyWithAuth(protocol)) {
+        provider.setCredentials(
+            new AuthScope(
+                System.getProperty(protocol + ".proxyHost"),
+                Integer.parseInt(System.getProperty(protocol + ".proxyPort"))),
+            new UsernamePasswordCredentials(
+                System.getProperty(protocol + ".proxyUser"),
+                System.getProperty(protocol + ".proxyPassword").toCharArray()));
+      } else if (systemEnvIsSetForProtocol(protocol)) {
+        CustomProxy proxyData = getProxySettingsFromEnvVar(protocol);
+        provider.setCredentials(
+            new AuthScope(proxyData.host, proxyData.port),
+            new UsernamePasswordCredentials(proxyData.username, proxyData.password.toCharArray()));
+      }
+    } catch (NumberFormatException e) {
+      throw new ConnectorInputException(
+          "Invalid proxy port: " + (System.getProperty(protocol + ".proxyPort")), e);
+    }
+    return provider;
+  }
+
+  public static HttpHost getProxyHost(String uncheckedProtocol, String requestUri) {
+    String protocol = fallbackOnHttpForInvalidProtocol(uncheckedProtocol);
+    try {
+      if (isValidProxy(protocol)) {
+        return new HttpHost(
+            protocol,
+            System.getProperty(protocol + ".proxyHost"),
+            Integer.parseInt(System.getProperty(protocol + ".proxyPort")));
+      } else if (systemEnvIsSetForProtocol(protocol)
+          && !doesTargetMatchNonProxy(protocol, requestUri)) {
+        var proxyData = getProxySettingsFromEnvVar(protocol);
+        return new HttpHost(protocol, proxyData.host, proxyData.port);
+      }
+    } catch (NumberFormatException e) {
+      throw new ConnectorInputException(
+          "Invalid proxy port: " + (System.getProperty(protocol + ".proxyPort")), e);
+    }
+    return null;
+  }
+
+  private static boolean doesTargetMatchNonProxy(String protocol, String requestUri) {
+    String nonProxyHostsEnvVar =
+        System.getenv("CONNECTOR_" + protocol.toUpperCase() + "_PROXY_NON_PROXY_HOSTS");
+    return (nonProxyHostsEnvVar != null
+        && Arrays.stream(nonProxyHostsEnvVar.split("\\|"))
+            .anyMatch(
+                nonProxyHost ->
+                    requestUri.matches(nonProxyHost.replace(".", "\\.").replace("*", ".*"))));
+  }
```

Comment:
This check should be done against a proxy config without requiring to fetch the data from the environment. Possibly using a dedicated `ProxyConfig` as mentioned above.